File: frontend/src/types/historicalBackfill.ts
export interface HistoricalSeasonCoverage {
  schedule_events_seen: number;
  final_games: number;
  final_games_with_boxscores: number;
  final_games_ingested: number;
  missing_boxscores: number;
  failed_boxscores: number;
  open_identity_issues: number;
  open_quality_issues: number;
  game_completeness: string;
  game_coverage_window_id: number | null;
}

export interface HistoricalRangeSeason {
  season: string;
  status: string;
  season_run_id: number | null;
  started_at: string;
  finished_at: string;
  coverage: HistoricalSeasonCoverage | null;
  error_type: string | null;
  error_message: string | null;
}

export interface HistoricalRangeBackfill {
  run_id: number;
  sport_slug: string;
  start_season: string;
  end_season: string;
  status: string;
  boxscore_delay_seconds: number;
  resumed: boolean;
  started_at: string;
  finished_at: string;
  seasons_total: number;
  seasons_attempted: number;
  seasons_skipped: number;
  seasons_succeeded: number;
  seasons_partial: number;
  seasons_failed: number;
  seasons: HistoricalRangeSeason[];
}

export interface HistoricalRangeRequest {
  startSeason: string;
  endSeason: string;
  boxscoreDelaySeconds: number;
  resumeRunId?: number;
}

File: backend/app/schemas/historical_backfill.py
"""Schemas for bounded historical WBB season backfills."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.current_season import CurrentSeasonSyncRead
from app.schemas.season_stats import CumulativeStatsImportRead


class HistoricalSeasonCoverageRead(BaseModel):
    """Coverage evidence for one historical WBB season."""

    model_config = ConfigDict(from_attributes=True)

    schedule_events_seen: int
    final_games: int
    final_games_with_boxscores: int
    final_games_ingested: int
    missing_boxscores: int
    failed_boxscores: int
    open_identity_issues: int
    open_quality_issues: int
    game_completeness: str
    game_coverage_window_id: int


class HistoricalSeasonBackfillRead(BaseModel):
    """Operator-readable outcome for one bounded historical season backfill."""

    model_config = ConfigDict(from_attributes=True)

    run_id: int
    sport_slug: str
    season: str
    status: str
    started_at: datetime
    finished_at: datetime
    game_sync: CurrentSeasonSyncRead
    season_stats_status: str
    season_stats_error: str | None
    season_stats: CumulativeStatsImportRead | None
    coverage: HistoricalSeasonCoverageRead
